index.js

```diff
@@ -737,9 +737,21 @@ module.exports = class Hypercore extends EventEmitter {
       end = start + 1
     }
 
-    if (start >= end) return
+    const cleared = (opts && opts.diff) ? { blocks: 0 } : null
+
+    if (start >= end) return cleared
+
+    const before = cleared ? await Info.bytesUsed(this.core.blocks.storage) : null
 
     await this.core.clear(start, end)
+
+    const after = cleared ? await Info.bytesUsed(this.core.blocks.storage) : null
+
+    if (cleared) {
+      cleared.blocks = before - after
```

Comment:
lets Math.max this with 0, as the disk is not lockable

Reply:
There is no local lock that we could use? So within the same program works as expected

Reply:
good point, we should consider moving this into the locked section actually. Still needs the guard, as we dont own the disk.